Jenkinsfile

```diff
@@ -2,20 +2,48 @@
 found at https://github.com/ihmeuw/vivarium_build_utils
 Due to Jenkins convention, importable modules must be stored
 in the 'vars' folder.
-Jenkins needs to be configured globally to use the correct branch.
-To configure the repo/branch go to:
-* Manage Jenkins
-  * Configure System
-    * Global Pipeline Libraries section
-      * Library subsection
-        * Name: The Name for the lib
-        * Version: The branch you want to use. Throws an error
-                   for nonexistent branches.
-        * Project Repository: Url to the shared lib
-        * Credentials: SSH key to access the repo
 
 Updating the shared repo will take affect on the next pipeline invocation.
-The "_" denotes that all modules will be imported from the shared library.
 */ 
-@Library("vivarium_build_utils") _
-reusable_pipeline(scheduled_branches: ["main"], upstream_repos: ["layered_config_tree"])
+
+// Determine the vivarium_build_utils version and load the library
+library "vivarium_build_utils@${getVBUVersion()}"
+
+// Run the reusable pipeline
+reusable_pipeline(
```

Comment:
this hypothetically remains the only things that need to be changed on a per-repo bases.

The Jenkinsfile is obviously more complicate now but:
1. I've kept the number of code lines _above_ this call the same. The library call itself has changed but it's still just one line.
2. I at least moved all of the extra code down below in a Helper functions section.
3. The getVBUVersion function is downloading a python script from vbu main and so it's version-controlled in a single place.